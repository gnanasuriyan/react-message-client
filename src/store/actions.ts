export { alertHide, alertShow } from './slices/alerts';
export { setAppOptions } from './slices/app';
export { getMessageList, getMessageListFailure, getMessageListSuccess, postMessage, postMessageSuccess, postMessageFailure } from './slices/message';
export { login, loginSuccess, logOut, logOutSuccess } from './slices/user';
export { signup, signupSuccess, signupFailure } from './slices/signup';
