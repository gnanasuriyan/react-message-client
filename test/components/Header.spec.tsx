import React from 'react';
import { fireEvent, render, screen } from 'test-utils';

import { logOut } from '~/actions';

import Header from '~/components/Header';

const mockDispatch = vi.fn();

describe('Header', () => {
  it('should render properly', () => {
    render(<Header isAuthenticated={true} name={"some name"}/>);

    expect(screen.getByRole('banner')).toMatchSnapshot();
  });

  it('should handle clicks', async () => {
    render(<Header isAuthenticated={true} name={"some name"} />, { mockDispatch });
    fireEvent.click(screen.getByRole('button')!);

    expect(mockDispatch).toHaveBeenCalledWith(logOut());
  });
});
