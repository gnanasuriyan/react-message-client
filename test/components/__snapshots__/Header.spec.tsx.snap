// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render properly 1`] = `
<header
  class="css-157to9t"
  data-component-name="Header"
>
  <div
    class="css-1ejvhcb"
    data-component-name="Container"
  >
    <div
      class="css-29z1qi"
    >
      <h3
        class="css-1accte4"
        data-component-name="H3"
      >
        some name
      </h3>
    </div>
    <button
      class="css-17rt720"
      data-component-name="Logout"
    >
      <span
        class="css-lo0v0o"
        data-component-name="Text"
      >
        logout
      </span>
      <span
        class="css-10mt42n"
        data-component-name="Icon"
        name="sign-out"
      >
        <svg
          height="16"
          src="https://files.gilbarbara.dev/icons/sign-out.svg"
          title="sign-out"
          width="16"
        />
      </span>
    </button>
  </div>
</header>
`;
